docs/developer-guide/plugins.md

```diff
@@ -87,52 +86,55 @@ returns an observable of the response. It can be used to modify the request and/
-* A plugin may choose to respond to the request, in which case the request will not 
+* an interceptor "owns" the request until it passes it to the next interceptor, by
+  calling `chain.proceed`. At this point Styx calls `intercept` of the next interceptor.
+* If a plugin chooses to respond to the request, the request will not 
   get propagated any further down the pipeline.
 
-The intercept method is called with two arguments, a HttpRequest instance (self-explanatory), 
-and Chain. The Chain object represents a view of the Styx interceptor pipeline as for this 
-particular request. The `intercept()` method will use the Chain object, most importantly, to 
-pass the request object onwards in the pipeline. It can also be used to store some 
-request-specific context that is visible for other plugins.
+The `intercept` is called with two arguments: HTTP request, and a `Chain`. 
+The `Chain` represents a Styx interceptor pipeline as for this request, and
+it is used to pass the request on in the pipeline. Chain also stores request
+context attributes and exposes them to interceptors.
 
-The main structure of the intercept() method is always the same:
+The `intercept` always has the same structure:
 
-1. Transform HTTP request object in any way necessary.
-2. Call the chain.proceed() to pass the request forward in the pipeline. 
-   This returns a `StyxObservable<HttpResponse>`.
-3. Create a new HTTP response observable by applying any necessary 
-   transformations to the HTTP response observable obtained from a call to `chain.proceed()`.
-4. Return the new HTTP response observable.
+1. Transform HTTP request.
+2. Pass the request on by calling `chain.proceed`. 
+   This returns a `Eventual<LiveHttpResponse>`.
+3. Apply response transformations on the response `Eventual`.    
+4. Return an `Eventual` of the transformed response.
 
 Alternatively, when the interceptor responds to the request:
 
-1. Construct a response object.
-2. Return the response object wrapped inside a HTTP response Observable. Eg: `StyxObservable.of(response)`.
+1. Consume request body.
+2. Construct a response.
+3. Return an `Eventual` of the response.
 
 ### Plugin factory class
-A plugin factory class is responsible for creating an instance of the plugin. 
+
+A plugin factory class instantiates the plugin. 
 There is only one method to implement: `Plugin create(Environment environment);`
 
-The `create()` method can perform any necessary initialisation before starting 
-the plugin. Styx passes in an Environment object that contains plugin's 
+The `create` method performs any necessary initialisation before starting 
+the plugin. Styx passes in an `Environment` object that contains plugin's 
 configuration, a view of Styx configuration, and also a metrics registry.
 
-Styx will not start forwarding traffic until `create()` is successfully called 
+Styx will not start forwarding traffic until `create` is successfully called 
```

Comment:
Maybe instead of "until `create` is successfully called "
we should have "until the call to `create` returns successfully"?

Just a small distinction in meaning.

Reply:
ok